src/connection_manager.c

```diff
@@ -790,6 +790,8 @@ qd_error_t qd_entity_refresh_connector(qd_entity_t* entity, void *impl)
     int i = 1;
     int num_items = 0;
 
+    sys_mutex_lock(ct->lock);
+
```

Comment:
If the connector state is CXTR_STATE_DELETED, should we just return immediately? Why loop thru the conn_info_list and so on? When the user runs a qdmanage QUERY --type=connector, I think the deleted connector (CXTR_STATE_DELETED) should not even show up in the list of connectors.